src/psygnal/_group_descriptor.py

```diff
@@ -265,7 +265,12 @@ def _setattr_and_emit_(self: object, name: str, value: Any) -> None:
             group: SignalGroup | None = getattr(self, signal_group_name, None)
             signal: SignalInstance | None = getattr(group, name, None)
```

Comment:
with a dataclass `name` field, `getattr(group, name, None)` returns `''` because of the deprecated looking for `SignalRelay` attributes.
Then `signal` is a `str` , leading to errors

This works:
```python
if group is None or not isinstance(group, SignalGroup):
    return super_setattr(self, name, value)
signal: SignalInstance | None = group[name] if name in group else None
```

Reply:
hah, that's funny, I had [a previous iteration](https://github.com/pyapp-kit/psygnal/pull/269/commits/9f44dfcfbd40b640bec98bbb7fcfb8a0749e814c#diff-8218e71615e3cb1b30f78bd2143ca77661925d39178dd9ccad62dc0f62206c98R227-R228) that would have solved this (by first looking at the signal names in `SignalGroup.__getattr__` before checking if the name is on `SignalInstance`), but then removed it because I forgot why I had put it there.  THIS is why :joy:  I'll add it back and add a test

Reply:
> with a dataclass name field, getattr(group, name, None) returns '' because of the deprecated looking for SignalRelay attributes.

actually... @getzze, are you sure about that? If I run the following code on the commit you commented on 8eab8af, i don't get the issue you describe:

```python
from psygnal import evented, SignalGroup, SignalInstance
from dataclasses import dataclass


@evented
@dataclass
class Foo:
    name: str


obj = Foo("foo")
assert obj.name == "foo"
group = obj.events
assert isinstance(group, SignalGroup)
assert "name" in group
assert isinstance(group.name, SignalInstance)
assert group["name"] is group.name
```

I think the explanation is that `getattr(group, name, None)` first calls getattr on the SignalGroup *class* (not the instance), and that finds the Signal descriptor's `__get__` method, and it never reaches `__getattr__` method on the group instance (and you never reach the deprecation warning).  I did have a comment on this in the code afterall [here](https://github.com/pyapp-kit/psygnal/pull/269/files#diff-8218e71615e3cb1b30f78bd2143ca77661925d39178dd9ccad62dc0f62206c98R305-R311).  Did you try this locally and find otherwise?  Or were you just looking at the code online and expecting it to hit the deprecation?

Reply:
I've added back the following code to `__getattr__` just in case, but it will likely never hit:

```python
        # Note, technically these lines aren't actually needed because of the descriptor
        # protocol.  Accessing a name on the instance will first look in the
        # instance's __dict__, and then in the class's __dict__, which
        # will call Signal.__get__ and return the SignalInstance.
        # these lines are here as a reminder to developers.
        if name != "_psygnal_instances" and name in self._psygnal_instances:
            return self._psygnal_instances[name]  # pragma: no cover
```

Reply:
> actually... @getzze, are you sure about that? If I run the following code on the commit you commented on [8eab8af](https://github.com/pyapp-kit/psygnal/commit/8eab8afb7b309bac5aec1853276e2718d9bfc620), i don't get the issue you describe:

Maybe the dataclass field was named `_name` not `name`, I cannot test it now.